### PR TITLE
Common Utils: Remove instanceof

### PR DESCRIPTION
**Pull request by anthony-murphy**

### CODE REVIEW

File: common/lib/common-utils/src/bufferBrowser.ts [hunk 1]
```diff
@@ -44,10 +44,29 @@ export const stringToBuffer = (input: string, encoding: string): ArrayBufferLike
...
  */
 export class IsoBuffer extends Uint8Array {
+    private readonly ___isIsoBuffer___  = true;
+
+    public static isIsoBuffer(obj: any): obj is IsoBuffer {
+        const maybe = obj as IsoBuffer;
+        return maybe.___isIsoBuffer___ === true;
+    }
+
     // Need to have ctor for it to be in proto chain for instanceof check in from() method to work
```
Comment (vladsud) [inline common/lib/common-utils/src/bufferBrowser.ts:70]: You can remove ctor below.

**Review by vladsud — APPROVED** [review]

File: common/lib/common-utils/src/bufferBrowser.ts [hunk 2]
```diff
@@ -44,10 +44,29 @@ export const stringToBuffer = (input: string, encoding: string): ArrayBufferLike
 export const bufferToString = (blob: ArrayBufferLike, encoding: string): string =>
      IsoBuffer.from(blob).toString(encoding);
 
+/**
+ * Determines if an object is an array buffer
+ * @param obj - The object to determine if it is an ArrayBuffer
+ */
+export function isArrayBuffer(obj: any): obj is ArrayBuffer {
```
Comment (curtisman) [inline common/lib/common-utils/src/bufferBrowser.ts:51]: Can we avoid this kind of check at all?

Comment (anthony-murphy) [reply common/lib/common-utils/src/bufferBrowser.ts:51]: No idea. @heliocliu , and @vladsud  have the most context here

File: common/lib/common-utils/src/bufferBrowser.ts (hunk 1 above)
Comment (anthony-murphy) [reply common/lib/common-utils/src/bufferBrowser.ts:70]: question. do we pass these across the loader container boundary? as this change won't be back compat

File: common/lib/common-utils/src/bufferBrowser.ts (hunk 2 above)
Comment (vladsud) [reply common/lib/common-utils/src/bufferBrowser.ts:51]: Yes we do.

Comment (vladsud) [reply common/lib/common-utils/src/bufferBrowser.ts:51]: I believe that combination of these 2 PRs being in same release (i.e. compiled into both client and loader) should be sufficient to address the bug:
https://github.com/microsoft/FluidFramework/pull/5035
https://github.com/microsoft/FluidFramework/pull/5173

Later has exactly same error in description.
The problem is that we did not picked up first PR before cutting release.

**Pull request closed without merging**